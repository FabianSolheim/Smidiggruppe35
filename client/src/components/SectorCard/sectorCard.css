.sector-container {
    background-color: #FFF;
    height: 420px;
    position: relative;
}
.sector-image{
    width: 100%;
    object-fit: contain;
}


@media screen and (min-width: 600px){
    .sector-container {
        box-shadow: 0px 4px 30px rgba(0,0,0,0.1);
    }
}@media screen and (min-width: 1280px){
    .sector-container {
        height: 480px;
    }
}
