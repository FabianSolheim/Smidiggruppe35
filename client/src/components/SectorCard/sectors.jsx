export const sectors = [
  {
    title: "Water",
    content:
      "Imagine spending hours everyday searching for water. \n" +
      "but with clean water, everything changes",
    image: "https://i.imgur.com/pi1w9X5.png",
    linkTo: "/water",
  },
  {
    title: "Knowledge",
    content:
      "Knowledge is one of the greatest tools for \n" +
      "change, one of the most important factors to \n" +
      "fight the climate crisis, and one of the best ways \n" +
      "to break the cycle of poverty",
    image: "https://i.imgur.com/ltNtgN8.png",
    linkTo: "/knowledge",
  },
];
