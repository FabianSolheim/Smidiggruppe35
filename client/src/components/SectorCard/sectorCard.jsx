import { Box, Card, Grid, Typography } from "@mui/material";
import CustomButton from "../CustomButton/CustomButton";
import "./sectors";
import "./sectorCard.css";

export function SectorCard({ title, content, image, linkTo }) {
  return (
    <Box textAlign="center" className="sector-container">
      <Grid container flexDirection="row">
        <Card sx={{ maxWidth: 500 }} className="sector-card">
          <img src={image} width={250} />
          <Typography className="sector-header">{title}</Typography>
          <Typography fontSize={15} className="sector-content">
            {content}{" "}
          </Typography>
          <Box alignItems="center" textAlign="center" className="sector-button">
            <CustomButton to={linkTo} variant="contained">
              Case study
            </CustomButton>
          </Box>
        </Card>
      </Grid>
    </Box>
  );
}
