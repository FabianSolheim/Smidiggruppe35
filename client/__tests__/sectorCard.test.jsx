import React from "react";
import ReactDOM from "react-dom";
import { SectorCard } from "../src/components/SectorCard/sectorCard";
import { MemoryRouter } from "react-router-dom";

const testData = [
  {
    title: "testTitle",
    content: "test content",
    image: "imageURL",
    linkTo: "testData",
  },
];

describe("sectorCard test", () => {
  it("shows sector cards", () => {
    const domElement = document.createElement("Box");
    ReactDOM.render(
      <MemoryRouter>
        {testData.map((test) => (
          <SectorCard
            key={test.title}
            title={test.title}
            content={test.content}
            image={test.image}
            linkTo={test.linkTo}
          />
        ))}
      </MemoryRouter>,
      domElement
    );
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
