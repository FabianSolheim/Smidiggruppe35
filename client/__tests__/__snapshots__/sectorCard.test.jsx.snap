// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sectorCard test shows sector cards 1`] = `"<a href=\\"/testData\\"><div class=\\"sector-container MuiBox-root css-i3nlsl\\"><div class=\\"MuiGrid-root MuiGrid-container css-1hqcngf-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><img src=\\"imageURL\\" class=\\"sector-image\\"></div><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\"><div class=\\"MuiBox-root css-1yuhvjn\\"><h3 class=\\"MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom sector-header css-1kn7tn9-MuiTypography-root\\">testTitle</h3><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter sector-content css-95p7p-MuiTypography-root\\">test content</p><div class=\\"sector-button MuiBox-root css-zw9846\\"><a class=\\" css-vcbgxq-MuiButton-root\\" component=\\"button\\" focusvisibleclassname=\\"\\" style=\\"padding: 12px 22px; text-decoration: none; font-weight: 500; border-radius: 0; text-transform: none;\\" classes=\\"[object Object]\\" href=\\"/testData\\">Case study</a></div></div></div></div></div></a>"`;
